src/bitmap.rs

```diff
@@ -37,11 +53,20 @@ impl<const PAGE_SIZE: usize> BitmapPageAllocator<PAGE_SIZE> {
 impl<const PAGE_SIZE: usize> BaseAllocator for BitmapPageAllocator<PAGE_SIZE> {
     fn init(&mut self, start: usize, size: usize) {
         assert!(PAGE_SIZE.is_power_of_two());
-        let end = super::align_down(start + size, PAGE_SIZE);
-        let start = super::align_up(start, PAGE_SIZE);
-        self.base = start;
+
+        // Range for real:  [align_up(start, PAGE_SIZE), align_down(start + size, PAGE_SIZE))
+        let start = crate::align_up(start, PAGE_SIZE);
+        let end = crate::align_down(start + size, PAGE_SIZE);
```

Comment:
Swap these two lines, `start` should be the one in the argument.

Reply:
sorry, it's a stupid mistake